### PR TITLE
feat: enable directory features and entitlements on directory level

### LINKED ISSUE #275: [FEATURE] Support Directory Features
When creating a directory several features like entitlements and authorizations can be specified via a feature section. 
This enables several organizational options on directory level like managing entitlements and quotas that will be propagated to the subaccounts.

We should enable that for the Terraform providers for the directory resource and the consequent resources like entitlements that need to be available for the directory level

This also needs to be validated from a usage perspective of the BTP CLI if this features are adopted to make the right priority.

### PR DESCRIPTION
**Pull request by lechnerc77**
## Purpose

* This PR enables the usage of directory features for the Terraform provider
* It introduces the resource for entitlements on directory level
* Closes #275 

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

## Other Information

Known gap:
- **import**: due to the platform API the `distribute` flag cannot be retrieved in an import. This must be manually adjusted when doing the resource import. 
- **enablement**: The BTP CLI offers an option to enable features via a dedicated command (see <https://help.sap.com/docs/btp/btp-cli-command-reference/btp-enable-accounts-directory>). This is not supported by the provider as we cannot distinguish what needs to be updated or need to make two calls to the backend including the error handling and rollback. There are workarounds to avoid this scenario with Terraform.

## Checklist for reviewer

The following organizational tasks must be completed before merging this PR:

* [X] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [X] The PR has the matching labels assigned to it.
* [X] The PR has a milestone assigned to it.
* [X] If the PR closes an issue, the issue is referenced.
* [X] Possible follow-up items are created and linked.

### CODE REVIEW

**Comment by lechnerc77** [comment]
Detected fixture drift is okay due to re-recording of fixture

File: internal/btpcli/facade_accounts_directory.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type DirectoryCreateInput struct {
 	Subdomain     *string             `btpcli:"subdomain"`
 	Labels        map[string][]string `btpcli:"labels"`
 	Globalaccount string              `btpcli:"globalAccount"`
-	//DirectoryAdmins string          `btpcli:"directoryAdmins"`
```
Comment (v0lkc) [inline internal/btpcli/facade_accounts_directory.go:36]: Did you remove this on purpose?

File: internal/provider/resource_directory.go [hunk 2]
```diff
@@ -368,3 +379,25 @@ func (rs *directoryResource) Delete(ctx context.Context, req resource.DeleteRequ
...
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "DEFAULT")
+	}
+
+	if slices.Contains(directoryFeatures, "ENTITLEMENTS") {
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "ENTITLEMENTS")
+	}
+
+	if slices.Contains(directoryFeatures, "AUTHORIZATIONS") {
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "AUTHORIZATIONS")
+	}
```
Comment (v0lkc) [inline internal/provider/resource_directory.go]: Could you please refactor this part and make the strings constant?

File: internal/tfutils/tfutils.go [hunk 3]
```diff
@@ -120,6 +124,13 @@ func setBool(field reflect.Value, tagValue string, out map[string]string) {
 	out[tagValue] = fmt.Sprintf("%v", fieldVal)
 }
 
+func setStringSlice(field reflect.Value, tagValue string, out map[string]string) {
+	if !field.IsNil() {
+		valueString := fmt.Sprintf("%v", strings.Join(field.Interface().([]string), ","))
+		out[tagValue] = valueString
+	}
+}
```
Comment (v0lkc) [inline internal/tfutils/tfutils.go:136]: Could you please add a small test case to capture the intended behaviour of this?

File: internal/btpcli/facade_accounts_directory.go (hunk 1 above)
Comment (lechnerc77) [reply internal/btpcli/facade_accounts_directory.go:36]: Yes this was the original state and I left it commented as this functionality is inconsistent compared to the subaccount (not possible to specify further admins )and inconsistent concerning the overall handling of assignment of role collections.
This resource should create/modify/delete the entity `directory`. Any role specifics shall be handled by the corresponding role collection assignment functionality

File: internal/provider/resource_directory.go (hunk 2 above)
Comment (lechnerc77) [reply internal/provider/resource_directory.go]: Switched to constants

File: internal/tfutils/tfutils.go (hunk 3 above)
Comment (lechnerc77) [reply internal/tfutils/tfutils.go:136]: Added test case for this scenario

**Pull request merged**